tree/dataframe/src/RLoopManager.cxx

```diff
@@ -315,6 +315,11 @@ void RLoopManager::RunTreeReader()
    while (r.Next() && fNStopsReceived < fNChildren) {
       RunAndCheckFilters(0, r.GetCurrentEntry());
    }
+   if (r.GetEntryStatus() != TTreeReader::kEntryNotFound && fNStopsReceived < fNChildren) {
+      // something went wrong in the TTreeReader event loop
+      throw std::runtime_error("An error was encountered while processing the data. TTreeReader status code is: " +
+                               std::to_string(r.GetEntryStatus()));
+   }
    CleanUpTask(0u);
```

Comment:
Does this cleanup need to be done before throwing the exception or it is okay to not run it at all?

Reply:
It's "ok" not to run it at all: no leaks or anything like that.

Without calling `CleanUpTask`, the computation graph is not cleared of (partially, wrongly) executed nodes, and the results are not "finalized" (e.g. partial results of different threads are not merged). With `CleanUpTask` called, the computation graph is cleared, and the results are finalized (but anyway wrong, because something went wrong during the event loop).

I would recommend users throw away the dataframe altogether if an exception is thrown during the event loop (that's _if_ they catch the exception at all), so whether `CleanUpTask` is called or not does not matter.

Reply:
I'm confusing `CleanUpTask` and `CleanUpNodes` a little bit in my answer above when I describe what `CleanUpTask` does, but since `CleanUpNodes` is also not executed, the answer mostly stands

Reply:
Well -- cannot hurt to call `CleanUpTask` and might actually prevent tear-down issues in `SnapshotHelperMT`. Will fix :smile: Thanks for pointing this out

Reply:
@pcanal turns out this deserved a whole separate PR. It is now #5214

Reply:
I take it that question is resolved now.

Reply:
Yes, resolved by #5214 (just merged)